### PR TITLE
vfs caching

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
Implement meta-data caching option in FileSystemLike.
Delete class FileSystemResource.FileInfo and replace usages with caching FileSystemLike

#### Related Pull Requests
- 

#### Changes
-

### CODE REVIEW

File: api/src/org/labkey/vfs/vfsTestCase.jsp [hunk 1]
```diff
@@ -205,6 +223,63 @@
 
     }
 
+   void testCaching(FileSystemLike fs) throws Exception
+    {
+        // Modify the local fs directly and check that FS does not pickup changes immediately
+        fs = fs.getCachingFileSystem();
+        FileLike root = fs.getRoot();
+        File ROOT = root.toNioPathForWrite().toFile();
```
Comment (labkey-jeckels) [inline api/src/org/labkey/vfs/vfsTestCase.jsp:231]: Is there a reason to use non-standard casing for these variables?

File: pipeline/src/org/labkey/pipeline/api/PipeRootImpl.java [hunk 2]
```diff
@@ -435,6 +425,30 @@ public Pair<File,File> _resolveRoot(org.labkey.api.util.Path path)
         return new Pair<>(root,file);
     }
 
+//    /** resolve a path treating it as an absolute file system path */
+//    @Nullable
+//    private Pair<FileLike,FileLike> _resolveRootRelative(org.labkey.api.util.Path path)
```
Comment (labkey-jeckels) [inline pipeline/src/org/labkey/pipeline/api/PipeRootImpl.java]: Worth committing or deleting?

**Review by labkey-jeckels — APPROVED** [review]

File: api/src/org/labkey/vfs/vfsTestCase.jsp (hunk 1 above)
Comment (labkey-matthewb) [reply api/src/org/labkey/vfs/vfsTestCase.jsp:231]: just trying to mentally pair up the matching FileLike and File objects

File: pipeline/src/org/labkey/pipeline/api/PipeRootImpl.java (hunk 2 above)
Comment (labkey-matthewb) [reply pipeline/src/org/labkey/pipeline/api/PipeRootImpl.java]: That is for future use (PipeRootImpl conversion).  I'll clean that up.

**Pull request merged**